library/kani/src/invariant.rs

```diff
@@ -14,84 +15,33 @@ use std::num::*;
 ///
 /// This trait is unsafe since &self might represent an invalid value. The `is_valid()` function
 /// must return `true` if and only if the invariant of its type is held.
+///
+/// # Deprecation
+///
+/// We have decided to deprecate this trait in favor of using `Arbitrary` for now. The main
+/// benefit of using Invariant was to avoid calling `kani::any_raw()` followed by `kani::assume()`.
+/// However, `kani::any_raw()` today cannot guarantee the rust type validity invariants, which
+/// could lead to UB in our analysis.
+#[deprecated(
+    since = "0.8.0",
+    note = "With `kani::Invariant`, Kani cannot guarantee that the type respects the language type invariants which may trigger UB. Use `kani::Arbitrary` instead. \
```

Comment:
Whoops, this got messed up due to my comment. You can remove the next line and perhaps split this line into two.